src/io/parsers/VolumeModel.js

```diff
@@ -109,11 +112,38 @@ class VolumeModel {
   }
 
   _getVolumeInfo() {
-    return _.pick(this._header, ['dmean', 'dmin', 'dmax', 'sd']);
+    return _.pick(this._header, ['dmean', 'dmin', 'dmax', 'sd', 'angles', 'delta']);
+  }
+
+  _setBoxParams(xaxis, yaxis, zaxis) {
+    // if axes are not orthogonal, the origins might not match with box coordinates - need to make shift
+    let shiftX = 0;
+    let shiftY = 0;
+
+    if (this._header.angles.z >= Math.PI / 2) {
+      shiftX += Math.abs(yaxis.x);
+    }
+    if (this._header.angles.y >= Math.PI / 2) {
+      shiftX += Math.abs(zaxis.x);
+    }
+    if (this._header.angles.x >= Math.PI / 2) {
+      shiftY += Math.abs(zaxis.y);
+    }
+
+    this._header.delta.XY = Math.abs(yaxis.x) / (Math.abs(xaxis.x) + Math.abs(yaxis.x) + Math.abs(zaxis.x));
```

Comment:
Speaking about CodeClimate issues in these lines... Is denominator here just a `this._boxSize.x` calculated below? Probably it will not look like a code duplication if the formula is simpler. Also, think about vectorizing the formula too (what do these formulas mean?).

Reply:
Your comment was helpful, thank you. Refactored.